### PR TITLE
Use server timestamp for setting JWT expiry

### PR DESCRIPTION
**Pull request by pkandarpa-cs**

### CODE REVIEW

File: cortex/utils.py [hunk 1]
```diff
@@ -152,11 +165,18 @@ def generate_token(config, validity=2):
...
             "sub": config.get("username"),
         }
+
+        server_ts_dt = datetime.fromtimestamp(
+            _get_fabric_server_ts(config) / 1000, tz=tz.gettz("UTC")
+        )  # fabric info returns serverTs in milliseconds
+
+        expiry = server_ts_dt + timedelta(minutes=validity)
+
         token = py_jwt.generate_jwt(
```
Comment (jgielstra-cs) [inline cortex/utils.py:175]: Missing issueAt / IAT ?

Comment (pkandarpa-cs) [reply cortex/utils.py:175]: Does this go into `other_headers` ?

File: cortex/utils.py [hunk 2]
```diff
@@ -152,11 +165,18 @@ def generate_token(config, validity=2):
             "aud": config.get("audience"),
             "sub": config.get("username"),
         }
+
+        server_ts_dt = datetime.fromtimestamp(
```
Comment (jgielstra-cs) [inline cortex/utils.py:169]: Not a big deal but 
datetime.fromtimestamp(timestamp, tz=timezone.utc).

^^ Can use this over adding a new dependency ..

File: cortex/utils.py (hunk 1 above)
Comment (jgielstra-cs) [reply cortex/utils.py:175]: Compared to what we do in cli
```
    return new jose.SignJWT({})
        .setProtectedHeader({ alg: 'EdDSA', kid: jwk.kid })
        .setSubject(username)
        .setAudience(audience)
        .setIssuer(issuer)
        .setIssuedAt(Math.floor(serverTs / 1000)) // in seconds
        .setExpirationTime(expiry)
        .sign(jwtSigner, { kid: jwk.kid });
```

Think it ends up payload...  should be toplevel claim ...  https://pyjwt.readthedocs.io/en/stable/usage.html?highlight=generate#registered-claim-names. looking

Comment (pkandarpa-cs) [reply cortex/utils.py:175]: ok looks py_jwt sets this automatically

Comment (jgielstra-cs) [reply cortex/utils.py:175]: claims['iat'] = timegm(now.utctimetuple()). Lame...  so if client's is  ahead/fast we'll have token issues

Comment (pkandarpa-cs) [reply cortex/utils.py:175]: yeah im setting it to serverts in the payload

Comment (pkandarpa-cs) [reply cortex/utils.py:177]: done

File: cortex/utils.py (hunk 2 above)
Comment (pkandarpa-cs) [reply cortex/utils.py]: done

**Review by yreddy-CS — APPROVED** [review]

**Review by jgielstra-cs — APPROVED** [review]

**Pull request merged**